### PR TITLE
implement button and logic for canceling reservation request

### PR DESCRIPTION
**Pull request by lian-jung**
## Summary by Sourcery

Add end-to-end support for cancelling reservation requests from the UI through application services and GraphQL, and enhance reservation-related UI interactions and tests.

New Features:
- Introduce a CancelReservation mutation and application-service command to allow users to cancel reservation requests.
- Add a shared CancelReservationPopconfirm UI component and reusable popconfirm testing utilities for consistent cancellation UX.
- Wire listing and reservation UI components to call the cancellation mutation, including loading states and success/error messaging.

Enhancements:
- Expand Storybook stories and interaction tests for listing information and reservation actions to cover reservation creation, cancellation, loading, and error paths.
- Refine ReservationActions and ListingInformation button behavior to use confirmation dialogs for cancellations and to better reflect available actions by status.
- Improve reservation-request resolver test coverage with new scenarios for authentication, missing users, and cancellation outcomes.
- Tidy Vitest coverage configuration and centralize Ant Design loading indicator detection in UI tests.

Tests:
- Add BDD-style application-service tests and feature specs for the new cancel reservation command, including edge cases and authorization failures.
- Add Storybook stories that exercise reservation creation/cancellation flows and popconfirm behavior across various UI components.

Chores:
- Align react-router-dom versions across packages to a newer compatible release.

### CODE REVIEW

**Comment by lian-jung** [comment]
@sourcery-ai review

**Comment by lian-jung** [comment]
@sourcery-ai review

**Comment by lian-jung** [comment]
@sourcery-ai review

**Comment by lian-jung** [comment]
@sourcery-ai review

**Comment by lian-jung** [comment]
@sourcery-ai review

**Comment by lian-jung** [comment]
@sourcery-ai review

**Comment by lian-jung** [comment]
@sourcery-ai review

**Comment by lian-jung** [comment]
@sourcery-ai review

**Comment by lian-jung** [comment]
@sourcery-ai review

**Comment by lian-jung** [comment]
@sourcery-ai review

**Comment by lian-jung** [comment]
@sourcery-ai review

File: packages/sthrift/application-services/src/contexts/reservation-request/reservation-request/cancel.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+			| undefined;
+		await dataSources.domainDataSource.ReservationRequest.ReservationRequest.ReservationRequestUnitOfWork.withScopedTransaction(
+			async (repo) => {
+				const reservationRequest = await repo.getById(command.id);
+				if (!reservationRequest) {
+					throw new Error('Reservation request not found');
+				}
+
+				const reserver = await reservationRequest.loadReserver();
+				if (reserver.id !== command.callerId) {
```
Comment (jasonmorais) [inline packages/sthrift/application-services/src/contexts/reservation-request/reservation-request/cancel.ts]: you do this check in the setter that updates the field in the aggregate not at the application service level

**Comment by lian-jung** [comment]
@sourcery-ai review

**Comment by lian-jung** [comment]
@sourcery-ai review

File: .snyk
Comment (jasonmorais) [inline .snyk:1]: For snyk, look at this pr for what was required in this file https://github.com/simnova/sharethrift/pull/338/files

anything else i believe was upgradable, my pr is good for vulnerabilities

File: apps/ui-sharethrift/src/components/layouts/home/components/view-listing/listing-information/listing-information.tsx [hunk 2]
```diff
@@ -63,6 +65,7 @@ export const ListingInformation: React.FC<ListingInformationProps> = ({
 	reservationDates,
 	onReservationDatesChange,
 	reservationLoading = false,
+	cancelLoading = false,
```
Comment (jasonmorais) [inline apps/ui-sharethrift/src/components/layouts/home/components/view-listing/listing-information/listing-information.tsx:68]: Is there a reason to seperate loading between data and cancel, or could they be combined into a single value, do they need to be passed into this component or could that be handled by componentqueryloader? i would think about those two things depending on the usecase

File: packages/sthrift/ui-components/src/test-utils/popconfirm-test-utils.ts
Comment (jasonmorais) [inline packages/sthrift/ui-components/src/test-utils/popconfirm-test-utils.ts:1]: How many places is this actually used? If its just 1-2/ maybe 3  places we dont want to extract logic out.

**Review by jasonmorais — COMMENTED** [review]
Few small comments
